app/commands/github/issue/search.rb

```diff
@@ -0,0 +1,75 @@
+module Github
+  class Issue::Search
+    include Mandate
+
+    # Use class method rather than constant for
+    # easier stubbing during testing
+    def self.requests_per_page
+      20
+    end
+
+    def initialize(action: nil, knowledge: nil, module_: nil, size: nil, type: nil, repo: nil, order: nil, page: 1)
+      @action = action
+      @knowledge = knowledge
+      @module_ = module_
+      @size = size
+      @type = type
+      @repo = repo
+      @order = order
+      @page = page
+    end
+
+    def call
+      @issues = Github::Issue
+
+      filter_repo!
+      filter_unclaimed!
+      filter_action!
+      filter_knowledge!
+      filter_module!
+      filter_size!
+      filter_type!
+
+      sort!
+      paginate!
+    end
+
+    private
+    attr_reader :action, :knowledge, :module_, :size, :type, :repo, :order, :page, :issues
+
+    def filter_repo!
+      return if repo.blank?
+
+      @issues = @issues.where(repo: repo)
```

Comment:
Issues are not linked to a track, but are instead linked to a repo (e.g. `exercism/ruby`). I could replace the `repo` argument with a `track` parameter, but with the current setup we can filter on issues for other repos to (like configlet).

Reply:
We need some wider functionality here, because if someone searches for "Ruby" I want to include *all* the ruby repo_urls (for main repo, tooling, etc). So I think we're better off tagging each **issue** with a `track_id` and then filtering on that.

I think we can **also** filter on `repo` but I think that should be an additional, seperate option to searching by track_id. Also, I think the param (coming in at least) should be `repo_url`.